olive/cli/auto_opt.py

```diff
@@ -0,0 +1,230 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+import json
+import logging
+import tempfile
+from argparse import ArgumentParser
+from copy import deepcopy
+from pathlib import Path
+from typing import ClassVar, Dict
+
+from olive.cli.base import BaseOliveCLICommand, add_remote_options, is_remote_run
+from olive.common.utils import set_tempdir
+
+logger = logging.getLogger(__name__)
+
+
+EVALUATE_TEMPLATE = {
+    "common_evaluator": {
+        "metrics": [
+            {
+                "name": "accuracy",
+                "type": "accuracy",
+                "sub_types": [
+                    {"name": "accuracy_score", "priority": 1, "goal": {"type": "max-degradation", "value": 0.01}},
+                    {"name": "f1_score"},
+                    {"name": "auroc"},
+                ],
+                "data_config": "data_config",
+            },
+            {
+                "name": "latency",
+                "type": "latency",
+                "sub_types": [
+                    {"name": "avg", "priority": 2, "goal": {"type": "percent-min-improvement", "value": 20}},
+                    {"name": "max"},
+                    {"name": "min"},
+                ],
+                "data_config": "data_config",
+                "user_config": {"io_bind": True},
+            },
+        ]
+    }
+}
+
+TEMPLATE = {
+    "input_model": {"type": "HfModel"},
+    "auto_optimizer_config": {},
+    "search_strategy": {"execution_order": "joint", "search_algorithm": "tpe", "num_samples": 5, "seed": 0},
+    "systems": {
+        "local_system": {
+            "type": "LocalSystem",
+            "accelerators": [{"device": "cpu", "execution_providers": ["CPUExecutionProvider"]}],
+        }
+    },
+    "host": "local_system",
+    "evaluators": EVALUATE_TEMPLATE,
+    "evaluator": "common_evaluator",
+    "target": "local_system",
+}
+
+
+class AutoOptCommand(BaseOliveCLICommand):
+    allow_unknown_args: ClassVar[bool] = True
+
+    @staticmethod
+    def register_subcommand(parser: ArgumentParser):
+        sub_parser = parser.add_parser(
+            "auto-opt",
+            help=("Automatically performance optimize input model"),
+        )
+
+        # model options
+        model_group = sub_parser.add_argument_group("model options")
+        model_group.add_argument(
+            "--model",
+            required=True,
+            help="Onnx input model path.",
+        )
+        model_group.add_argument(
+            "--task",
+            type=str,
+            default=None,
+            help="The task of the input model. Need to be specified if the input model is downloaded from huggingface.",
+        )
+        model_group.add_argument(
+            "--model_type",
```

Comment:
Let's avoid overloading "model_type". May be "model_framework"? or "model_kind"? or something else?

Reply:
adopted `model_framework`